ibm/config.go

```diff
@@ -1732,6 +1743,18 @@ func (c *Config) ClientSession() (interface{}, error) {
 	}
 	session.icdServiceAPI = icdAPI
 
```

Comment:
Add support to construct the private endpoint if the visibility is set to private